catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -96,23 +85,45 @@
 
     private static final int MAX_MEMORY_SIZE = 100 * MB;
 
+    private final Function<Destination, Optional<DestinationSink>> destinationSinkFactory;
+
     private OrderContents order;
 
     private AccessManagerImpl accessManager;
 
     private CatalogFramework catalogFramework;
 
-    private String protocol;
-
-    private int port;
-
-    public OrderRequestImpl(OrderContents order, String protocol, int port,
-            AccessManagerImpl accessManager, CatalogFramework catalogFramework) {
+    /**
+     * This constructor is only intended for unit testing.
+     */
+    OrderRequestImpl(OrderContents order, AccessManagerImpl accessManager,
+            CatalogFramework catalogFramework,
+            Function<Destination, Optional<DestinationSink>> destinationSinkFactory) {
         this.order = order;
-        this.protocol = protocol;
-        this.port = port;
         this.accessManager = accessManager;
         this.catalogFramework = catalogFramework;
+        this.destinationSinkFactory = destinationSinkFactory;
+    }
+
+    public OrderRequestImpl(OrderContents order, String protocol, int port,
+            AccessManagerImpl accessManager, CatalogFramework catalogFramework,
+            EmailConfiguration emailConfiguration) {
+        this(order, accessManager, catalogFramework, destination -> {
+            /*
+             * Uses CORBA exception structure to determine type of destination stored in incoming
+             * Destination object.
+             */
+            try {
+                return Optional.of(new FtpDestinationSink(destination.f_dest(), port, protocol));
+            } catch (BAD_OPERATION e) {
```

Comment:
destination.discriminator() will tell you what type of destination is configured. The isFTP() and isEmail() methods do this.